GameOfLife/Bipeds.py

```diff
@@ -0,0 +1,101 @@
+from pprint import pprint
+
+
+class Human():
+    def __init__(self, name, birthday, gender):
+        self.name = name
+        self.birthday = birthday
+        self.gender = gender
+
+    def generation(self):
+        if self.birthday in range(1990, 2000):
+            return "Gen X"
+        else:
+            return "Not Gen X"
+
+
+class Population:
+    def __init__(self, humans):
+        self.pop_list = humans
+
+    def filter_gender(self, value):
+        fgend_list = []
+        value = value.lower()
+        for people in self.pop_list:
+            if people.gender.lower() == value:
+                fgend_list.append(people.name)
+        fgend_list.sort()
+        return fgend_list
+
+    def filter_generation(self, value):
+        """
+        filter_generation must go through all humans in population and make a new list of humans to return if they match the given generation.
+        It has one generation
+        """
+        value = value.lower()
+        genx_list = []
+        # 1965-80
+        mill_list = []
+        # 1981-96
+        genz_list = []
+        # 1997-12
+        gena_list = []
+        # 2010-20
+        if "x" in value:
+            for people in self.pop_list:
+                if people.birthday in range(1965, 1980):
+                    genx_list.append(people.name)
+                    return genx_list
+
+            else:
+                return "There is no Gen X in the given list!"

```

Comment:
Dans cette fonction ci, il serait préférable de juste retourner la liste de monde vide.
Voici un example:
dans maya si je cherche a lister des geometries qui sont dans la scene pour iterer par dessus la commande `cmds.ls(type="mesh")` me retourne une liste de geometrie.
donc `for geo in cmds.ls(type='mesh'):` fonctionne mais s'il fallait que dans un cas ou la scene n'avait pas de geometries que la commande retourne une string plustot qu'une liste vide mon code pourrait etre vraiment brisé. retourne une liste vide quand yen a pas. 😉

Reply:
Fixed! :P